libs/wingsdk/src/fs/fs.ts

```diff
@@ -0,0 +1,279 @@
+ * The fs class is used for interacting with the file system.
+ * All file paths must be POSIX file paths (/ instead of \),
+ * and will be normalized to the target platform if running on Windows.
+ */
+export class Util {
+  /**
+   * Join all arguments together and normalize the resulting path.
+   */
+  public static join(...ps: string[]): string {
+    return path.join(...ps);
+  }
+  /**
+   * Return the directory name of a path.
+   * @param p The path to evaluate.
+   */
+  public static dirname(p: string): string {
+    return path.dirname(p);
+  }
+  /**
+   * Return the last portion of a path.
+   * @param p The path to evaluate.
+   */
+  public static basename(p: string): string {
+    return path.basename(p);
+  }
+  /**
+   * Solve the relative path from {from} to {to} based on the current working directory.
+   * At times we have two absolute paths, and we need to derive the relative path from one to the other.
+   */
+  public static relative(from: string, to: string): string {
+    return path.relative(from, to);
+  }
+  /**
+   * The right-most parameter is considered {to}. Other parameters are considered an array of {from}.
+   *
+   * Starting from leftmost {from} parameter, resolves {to} to an absolute path.
+   *
+   * If {to} isn't already absolute, {from} arguments are prepended in right to left order,
+   * until an absolute path is found. If after using all {from} paths still no absolute path is found,
+   * the current working directory is used as well. The resulting path is normalized,
+   * and trailing slashes are removed unless the path gets resolved to the root directory.
+   *
+   * @param ps A sequence of paths or path segments.
+   */
+  public static resolve(...ps: string[]): string {
+    return path.resolve(...ps);
+  }
+  /**
+   * Returns `true` if the path exists, `false` otherwise.
+   *
+   * @param p the path to evaluate.
+   */
+  public static exists(p: string): boolean {
+    const normPath = localePath(p);
+    return fs.existsSync(normPath);
+  }
+  /**
+   * Reads the contents of the directory.
+   */
+  public static readDir(dirpath: string): Array<string> {
+    const normPath = localePath(dirpath);
+    return fs.readdirSync(normPath);
+  }
+  /**
+   * Reads the contents of the directory if the path exists, `undefined` otherwise.
+   */
+  public static tryReadDir(dirpath: string): Array<string> | undefined {
+    if (!Util.exists(dirpath)) {
+      return undefined;
+    }
+    return Util.readDir(dirpath);
+  }
+  /**
+   * Create a directory.
+   * @param dirpath The path to the directory you want to create.
+   */
+  public static mkdir(dirpath: string, opts?: MkdirOptions): void {
+    const normPath = localePath(dirpath);
+    fs.mkdirSync(normPath, opts);
+  }
+  /**
+   * Returns the created directory path.
+   * Generates six random characters to be appended behind a required prefix to create a unique temporary directory.
+   */
+  public static mkdirTemp(prefix: string): string {
```

Comment:
Rename to `mkdtemp` and if `prefix` is undefined, then default to: `fs.mkdtempSync(path.join(os.tmpdir(), "wing"))` or something like that.

```suggestion
  public static mkdtemp(prefix?: string): string {
```

It's very useful to be able to just call `fs.mkdtemp()` and get a temp directory.